train.py

```diff
@@ -18,32 +19,33 @@
 from utilities.training_logger import TrainingLogger
 
 
-
-def train(params):
-    """Instantiate and train the HGN.
+def train(params, variational, dtype=torch.float):
+    """Instantiate and train the Hamiltonian Generative Network.
 
     Args:
-        params (dictionary): Experiment parameters (see experiment_params folder)
+        params (dict): Experiment parameters (see experiment_params folder).
+        variational (bool): Whether to perform variational or deterministic training.
+        dtype (torch.dtype): Data type to be used in tensor computations.
     """
-     # Set device
+    # Set device
     device = "cuda:" + str(
         params["gpu_id"]) if torch.cuda.is_available() else "cpu"
 
     # Pick environment
     env = EnvFactory.get_environment(**params["environment"])
 
     # Instantiate networks
-    encoder = EncoderNet(seq_len=params["rollout"]["seq_length"],
-                         in_channels=params["rollout"]["n_channels"],
-                         **params["networks"]["encoder"]).to(device)
+    encoder = EncoderNet(
+        seq_len=params["rollout"]["seq_length"], in_channels=params["rollout"]["n_channels"],
+        **params["networks"]["encoder"], dtype=dtype).to(device)
     transformer = TransformerNet(
         in_channels=params["networks"]["encoder"]["out_channels"],
-        **params["networks"]["transformer"]).to(device)
-    hnn = HamiltonianNet(**params["networks"]["hamiltonian"]).to(device)
+        **params["networks"]["transformer"], dtype=dtype).to(device)
+    hnn = HamiltonianNet(**params["networks"]["hamiltonian"], dtype=dtype).to(device)
     decoder = DecoderNet(
         in_channels=params["networks"]["transformer"]["out_channels"],
-        out_channels=params["rollout"]["n_channels"],
-        **params["networks"]["decoder"]).to(device)
+        out_channels=params["rollout"]["n_channels"], **params["networks"]["decoder"],
+        dtype=dtype).to(device)
```

Comment:
Is this in pep8 format?

Reply:
I think it is as in the [code layout section](https://www.python.org/dev/peps/pep-0008/#id16) they say to not put arguments in the first line when they can't fit in a single line.

We should actually define what is our limit line length.